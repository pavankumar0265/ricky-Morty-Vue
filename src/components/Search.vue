<template>
  <div class="mt-4 mr-8">
    <p class="ml-3">Search By Name</p>
    <input
      type="text"
      class="w-full p-2 border-2 border-black ml-3 lg:w-80"
      @input="handleChange"
    />
    <button
      class="rounded bg-slate-400 hidden m-2 p-2 h-10 lg:inline-block"
    >
      Search
    </button>
  </div>
</template>
<script setup>

import { computed } from "vue";
import { useStore } from "vuex";
const store = useStore();

function handleChange(e) {
  let rickMortyData = computed(() =>
    store.getters.setSearchedRickMorty(e.target.value)
  );
  store.commit("rickyMortyStateUpdating", rickMortyData.value);
  store.commit("setSearchedValues", e.target.value);
}
</script>
