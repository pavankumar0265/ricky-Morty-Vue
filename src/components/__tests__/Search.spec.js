import { describe ,it, expect } from 'vitest'
import Search from "../Search.vue"
import {mount} from '@vue/test-utils'
import { useStore } from "vuex";

describe('Search', () => {
    const wrapper = mount(Search)
    const store = useStore();
    it('should render correctly', () => {
        const wrapper = mount(Search)
        expect(wrapper.html()).toMatchSnapshot()
    })
    it('should check search correctly', () => {
        expect(wrapper.html()).toContain("Search")
    })
})