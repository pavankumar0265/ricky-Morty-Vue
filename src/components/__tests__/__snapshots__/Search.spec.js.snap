// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search > should render correctly 1`] = `
"<div class=\\"mt-4 mr-8\\">
  <p class=\\"ml-3\\">Search By Name</p><input type=\\"text\\" class=\\"w-full p-2 border-2 border-black ml-3 lg:w-80\\"><button class=\\"rounded bg-slate-400 hidden m-2 p-2 h-10 lg:inline-block\\"> Search </button>
</div>"
`;
